lib/topics_filter.rb

```diff
@@ -697,6 +720,15 @@ def filter_in(values:)
         @scope = base.merge(new_topics.or(unread_topics))
       end
 
+      if values.delete("unseen")
+        ensure_topic_users_reference!
+        first_seen_at = @guardian.user.first_seen_at || Time.at(0)
+        @scope =
+          @scope.where("topics.bumped_at >= ?", first_seen_at).where(
```

Comment:
Looking at `lib/topic_query.rb`, it seems like we have to consider whether the user is a whisperer or not to determine whether a topic is unseen or not.

https://github.com/discourse/discourse/blob/edeb9d6cfc36ea94957efbc016ebc9ad92e46e36/lib/topic_query.rb#L1260-L1265

For `new` and `unread`, we call `TopicQuery.new_filter` and `TopicQuery.unread_filter` respectively. Maybe we need to introduce an `TopicQuery.unseen_filter` instead?